enterprise_access/apps/api/v1/views.py

```diff
@@ -740,3 +748,58 @@ def partial_update(self, request, *args, **kwargs):
             properties=response.data
         )
         return response
+
+
+class AccessPolicyRedeemAPIView(APIView):
+    """
+    API to redeem a policy.
+    """
+    authentication_classes = [JwtAuthentication]
+    permission_classes = [permissions.IsAuthenticated]
+
+    @permission_required(
+        constants.REQUESTS_ADMIN_LEARNER_ACCESS_PERMISSION,
+        fn=lambda request: request.data.get('group_id')
```

Comment:
If the route for this view contains the policy uuid, this permission_required `fn` would need to change.  You could use something like what license-manager does for a similar use-case: https://github.com/openedx/license-manager/blob/b4e538da2bcd66bbf2638d0dfcaa2467e0a27a7d/license_manager/apps/api/v1/views.py#L1359-L1362
, which basically means creating a utility/helper function to trace through related objects to get the relevant enterprise customer uuid: https://github.com/openedx/license-manager/blob/b4e538da2bcd66bbf2638d0dfcaa2467e0a27a7d/license_manager/apps/api/utils.py#L100-L132